sway-lib-core/src/experimental/storage.sw

```diff
@@ -0,0 +1,9 @@
+library;
+
+/// Describes a location in storage specified by the `b256` key of a particular storage slot and an
+/// offset, in words, from the start of the storage slot at `key`. The parameter `T` is the type of 
+/// the data to be read from or written to at `offset`.
+pub struct StorageKey<T> {
+    key: b256,
```

Comment:
Maybe we should rename this to `slot`?

```suggestion
    slot: b256,
```

Perhaps this way we can more clearly distinguish between the terms "slot" and "key" throughout our code and documentation? E.g.

- **storage slot**: the `b256`-addressed storage slot.
- **storage key**: a location within a specific storage slot as described by `offset`.

Reply:
I will address this in a follow up PR where I'm adding the rest of the storage library.